src/PointwiseFunctions/Hydro/MassWeightedFluidItems.cpp

```diff
@@ -0,0 +1,90 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "MassWeightedFluidItems.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace hydro {
+
+template <typename DataType, size_t Dim, typename Frame>
+void u_lower_t(const gsl::not_null<Scalar<DataType>*> result,
+               const Scalar<DataType>& lorentz_factor,
+               const tnsr::i<DataType, Dim, Frame>& spatial_velocity_one_form,
+               const Scalar<DataType>& lapse,
+               const tnsr::I<DataType, Dim, Frame>& shift) {
+  result->get() = get(lorentz_factor) *
+                  (get(lapse) * (-1.0) +
+                   get(dot_product(spatial_velocity_one_form, shift)));
+}
+
+template <typename DataType>
+void mass_weighted_internal_energy(
+    const gsl::not_null<Scalar<DataType>*> result,
+    const Scalar<DataType>& tilde_d,
+    const Scalar<DataType>& specific_internal_energy) {
+  result->get() = get(tilde_d) * get(specific_internal_energy);
+}
+
+template <typename DataType>
+void mass_weighted_kinetic_energy(const gsl::not_null<Scalar<DataType>*> result,
+                                  const Scalar<DataType>& tilde_d,
+                                  const Scalar<DataType>& lorentz_factor) {
+  result->get() = get(tilde_d) * (get(lorentz_factor) - 1.0);
```

Comment:
I'd agree there are improvements to be made; the same is true for the Lorentz factor itself, which is currently calculated as
1/sqrt(1-v^2)
instead of
sqrt(1+u^2)
(with u^2 = gamma^{ij} u_i u_j and u the 4-velocity)

Reply:
The lorentz_factor ~ 1 case should indeed be unimportant here, so I'll leave it as is for now.